src/test/recovery/t/050_check_recovery_backup.pl

```diff
@@ -0,0 +1,158 @@
+
+	$primary->command_ok([ 'rm', '-rf', $backup_dir ],
+		"clear directory for backup '$backup_dir'");
+	mkdir($backup_dir);
+
+	$primary->command_ok([ 'pg_basebackup', '--target-gp-dbid', 2, '-D', "$backup_dir", '-p', $primary->port, '--verbose' ],
+		'pg_basebackup runs');
+	ok(-f "$backup_dir/PG_VERSION", "backup was created at '$backup_dir'");
+
+	$primary->safe_psql('test_db', "SELECT pg_create_restore_point('$restore_point')");
+}
+
+sub restore_backup_primary
+{
+	my ($backup_dir, $restore_options, $label) = @_;
+	my $primary_pgdata = $primary->data_dir;
+
+	print("Running: restore_backup_primary: '$restore_options' '$label'\n");
+
+	$primary->command_ok([ 'rm', '-rf', $primary_pgdata ],
+		"'$label': rm primary pgdata");
+	$primary->command_ok([ 'cp', '-r', $backup_dir, $primary_pgdata ],
+		"'$label': copy basebackup to primary");
+
+	# configure recovery
+	$primary->set_recovery_mode;
+	open my $conf, '>', "$primary_pgdata/postgresql.auto.conf";
+	print $conf "recovery_target_action = 'promote'\n";
+	print $conf "restore_command = 'cp -vr $archivedir/%f $primary_pgdata/%p'\n";
+	print $conf "$restore_options\n";
+	close $conf;
+	$primary->start;
+}
+
+sub wait_recovery_and_switch_wal_primary
+{
+	my ($label) = @_;
+
+	print("Running: wait_recovery_and_switch_wal_primary: '$label'\n");
+
+	$primary->poll_query_until('test_db', "SELECT true FROM pg_switch_wal()")
+		or die "'$label': Timed out while waiting for switch WAL after recovery";
+}
+
+sub restore_backup_standby
+{
+	my ($backup_dir, $restore_point, $primary_host, $primary_port, $label) = @_;
+
+	my $standby_pgdata = $standby->data_dir;
+	my $standby_port = $standby->port;
+
+	$standby->command_ok([ 'rm', '-rf', $standby_pgdata ],
+		"'$label': rm standby pgdata");
+	$standby->command_ok([ 'cp', '-r', $backup_dir, $standby_pgdata ],
+		"'$label': copy basebackup to standby");
+
+	$standby->set_recovery_mode;
+	open my $conf, '>', "$standby_pgdata/postgresql.auto.conf";
+	print $conf "recovery_target_action = 'shutdown'\n";
+	print $conf "restore_command = 'cp -vr $archivedir/%f $standby_pgdata/%p'\n";
+	print $conf "recovery_target_name = '$restore_point'\n";
+	close $conf;
+	$standby->adjust_conf('postgresql.conf', 'port', $standby_port);
+	$standby->start(fail_ok => 1);
+
+	$standby->wait_for_log('database system is shut down');
+
+	# configure primary connection with replication slot
+	unlink "$standby_pgdata/recovery.signal";
+	unlink "$standby_pgdata/postgresql.auto.conf";
+	$standby->set_standby_mode;
+	open $conf, '>', "$standby_pgdata/postgresql.auto.conf";
+	print $conf "primary_conninfo = ";
+	print $conf "'user=postgres port=" . $primary_port;
+	print $conf " host=" . $primary_host;
+	print $conf " sslmode=prefer sslcompression=1 gssencmode=prefer";
+	print $conf " krbsrvname=postgres target_session_attrs=any";
+	print $conf " application_name=gp_walreceiver'\n";
+	print $conf "primary_slot_name = 'internal_wal_replication_slot'\n";
```

Comment:
Why do we need all these settings if the tests pass successfully without them?

Reply:
Standby should correctly connect to standby after recovery. Replication slot is used for connect.

Reply:
Still, I think that only these parameters are sufficient for the test, the other parameters are unimportant.
```suggestion
	print $conf "primary_conninfo = 'port=$primary_port host=$primary_host'\n";
	print $conf "primary_slot_name = 'internal_wal_replication_slot'\n";
```

Reply:
Removed.